src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -57,13 +61,21 @@ def pileupSizeTask(self):
             rucioAuthUrl = self.rucioClient.rucioParams['auth_host']
             rucioHostUrl = self.rucioClient.rucioParams['rucio_host']
             rucioToken, _tokenValidity = getRucioToken(rucioAuthUrl, self.rucioAccount)
-            containers = [r['pileupName'] for r in docs]
+            containers = []
+            for doc in docs:
+                pileupName = doc.get('customName', '')
+                if pileupName == '':
+                    pileupName = doc['pileupName']
+                containers.append(pileupName)
             msg = f"Fetching pileup size for {len(containers)} containers "
             msg += f"against rucio url: {rucioHostUrl}"
             self.logger.info(msg)
             datasetSizes = getPileupContainerSizesRucio(containers, rucioHostUrl, rucioToken)
```

Comment:
Note that this function has a scope input parameter, which you are not using here and will fail things out for customName

Reply:
good catch, fixed.